Python/Product/PythonTools/PythonTools/Debugger/DebugAdapter/DebugAdapterProcess.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.Debugger.DebugAdapterHost.Interfaces;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.PythonTools.Debugger {
+    sealed class DebugAdapterProcess {
+        private readonly ITargetHostInterop _targetInterop;
+        private readonly string _debuggerAdapterDirectory;
+
+        private ITargetHostProcess _targetHostProcess;
+        private string _webBrowserUrl;
+
+        public DebugAdapterProcess(ITargetHostInterop targetInterop, string debuggerAdapterDirectory) {
+            _targetInterop = targetInterop;
```

Comment:
this class is not very useful anymore, it's only constructed to call `StartProcess`. You could merge `StartProcess` into its caller and `LaunchBrowserDebugger` moved there too.
I don't know what future changes, if any, are planned

Reply:
1. Will be updated in next commit. 

2. I prefer to keep it in a separate class because it's more organized and allows for cleaner integration of the upcoming changes (monitoring messages/events).

Reply:
If you're keeping it separate, then I would at least change debuggerAdapterDirectory to not be quoted when passed in. It is not something you would expect of a variable/field that holds a directory path. Instead I would quote it at the time you pass it to ExecuteCommandAsync, that's the thing that requires the quoted value.

Reply:
You're using `FirstOrDefault()` for passing in the python exe, which means it could be `null`. I wonder how `ExecuteCommandAsync` will behave in that case.

Reply:
1. Changed. 
2. How about putting a null/empty string check in place? If it is null/empty, show error box and return the ITargetHostProcess as null.

Reply:
Failed to launch debug adapter... sounds good enough